Frameworks/UIKit/UIBezierPath.mm

```diff
@@ -261,27 +251,23 @@ - (void)addClip {
  @Status Interoperable
 */
 - (void)stroke {
-    CGContextRef ctx = UIGraphicsGetCurrentContext();
-    CGContextSaveGState(ctx);
-    CGContextBeginPath(ctx);
-    CGContextAddPath(ctx, _workingPath);
-    CGContextSetLineWidth(ctx, _lineWidth);
-    CGContextSetLineJoin(ctx, _lineJoinStyle);
-    if (_patternCount > 0) {
-        CGContextSetLineDash(ctx, _dashPhase, &_pattern[0], _patternCount);
-    }
-    CGContextStrokePath(ctx);
-    CGContextRestoreGState(ctx);
+    [self strokeWithBlendMode:kCGBlendModeNormal alpha:1];
 }
 
 /**
  @Status Interoperable
 */
 - (void)setLineDash:(const CGFloat*)pattern count:(NSInteger)count phase:(CGFloat)phase {
+    if (!pattern || count == 0) {
+        _patternCount = 0;
```

Comment:
doesn't `_pattern.size()` make a good stand-in for keeping a separate _patternCount?